[2024-04-16T09:42:04.784+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-04-16T09:42:04.798+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spotify_final_dag.hello2 scheduled__2024-04-16T06:30:00+00:00 [queued]>
[2024-04-16T09:42:04.805+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spotify_final_dag.hello2 scheduled__2024-04-16T06:30:00+00:00 [queued]>
[2024-04-16T09:42:04.806+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 2
[2024-04-16T09:42:04.816+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): hello2> on 2024-04-16 06:30:00+00:00
[2024-04-16T09:42:04.823+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=3538) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-04-16T09:42:04.824+0000] {standard_task_runner.py:63} INFO - Started process 3542 to run task
[2024-04-16T09:42:04.824+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'spotify_final_dag', 'hello2', 'scheduled__2024-04-16T06:30:00+00:00', '--job-id', '896', '--raw', '--subdir', 'DAGS_FOLDER/dag.py', '--cfg-path', '/tmp/tmplfsj9u5s']
[2024-04-16T09:42:04.825+0000] {standard_task_runner.py:91} INFO - Job 896: Subtask hello2
[2024-04-16T09:42:04.861+0000] {task_command.py:426} INFO - Running <TaskInstance: spotify_final_dag.hello2 scheduled__2024-04-16T06:30:00+00:00 [running]> on host 52651c1f9987
[2024-04-16T09:42:04.928+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spotify_final_dag' AIRFLOW_CTX_TASK_ID='hello2' AIRFLOW_CTX_EXECUTION_DATE='2024-04-16T06:30:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-04-16T06:30:00+00:00'
[2024-04-16T09:42:04.929+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-04-16T09:42:08.479+0000] {logging_mixin.py:188} INFO - There was a problem with the connection. Trial number 1, Status Code: 401
[2024-04-16T09:42:11.748+0000] {logging_mixin.py:188} INFO - There was a problem with the connection. Trial number 2, Status Code: 401
[2024-04-16T09:42:11.750+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-04-16T09:42:11.751+0000] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/get_data.py", line 98, in processes
    album_data = get_album(connection())
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/get_data.py", line 11, in get_album
    for album in data["items"]:
                 ~~~~^^^^^^^^^
TypeError: 'NoneType' object is not subscriptable
[2024-04-16T09:42:11.761+0000] {taskinstance.py:1205} INFO - Marking task as UP_FOR_RETRY. dag_id=spotify_final_dag, task_id=hello2, execution_date=20240416T063000, start_date=20240416T094204, end_date=20240416T094211
[2024-04-16T09:42:11.814+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 896 for task hello2 ('NoneType' object is not subscriptable; 3542)
[2024-04-16T09:42:11.859+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-04-16T09:42:11.871+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-04-16T09:42:11.872+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2024-04-16T09:48:49.240+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-04-16T09:48:49.265+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spotify_final_dag.hello2 scheduled__2024-04-16T06:30:00+00:00 [queued]>
[2024-04-16T09:48:49.275+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spotify_final_dag.hello2 scheduled__2024-04-16T06:30:00+00:00 [queued]>
[2024-04-16T09:48:49.276+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 2
[2024-04-16T09:48:49.293+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): hello2> on 2024-04-16 06:30:00+00:00
[2024-04-16T09:48:49.304+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=4976) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-04-16T09:48:49.306+0000] {standard_task_runner.py:63} INFO - Started process 5001 to run task
[2024-04-16T09:48:49.306+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'spotify_final_dag', 'hello2', 'scheduled__2024-04-16T06:30:00+00:00', '--job-id', '1346', '--raw', '--subdir', 'DAGS_FOLDER/dag.py', '--cfg-path', '/tmp/tmp305_3siz']
[2024-04-16T09:48:49.307+0000] {standard_task_runner.py:91} INFO - Job 1346: Subtask hello2
[2024-04-16T09:48:49.369+0000] {task_command.py:426} INFO - Running <TaskInstance: spotify_final_dag.hello2 scheduled__2024-04-16T06:30:00+00:00 [running]> on host 52651c1f9987
[2024-04-16T09:48:49.483+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spotify_final_dag' AIRFLOW_CTX_TASK_ID='hello2' AIRFLOW_CTX_EXECUTION_DATE='2024-04-16T06:30:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-04-16T06:30:00+00:00'
[2024-04-16T09:48:49.484+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-04-16T09:48:49.911+0000] {logging_mixin.py:188} INFO - Connection successful
[2024-04-16T09:48:50.310+0000] {logging_mixin.py:188} INFO - Connection successful
[2024-04-16T09:48:50.748+0000] {logging_mixin.py:188} INFO - Connection successful
[2024-04-16T09:48:50.752+0000] {logging_mixin.py:188} INFO - Null Not Found
[2024-04-16T09:48:50.753+0000] {logging_mixin.py:188} INFO - Null Not Found
[2024-04-16T09:48:50.765+0000] {logging_mixin.py:188} INFO - Null Not Found
[2024-04-16T09:48:50.766+0000] {logging_mixin.py:188} INFO - Null Not Found
[2024-04-16T09:48:50.768+0000] {logging_mixin.py:188} INFO - Null Not Found
[2024-04-16T09:48:50.769+0000] {logging_mixin.py:188} INFO - Null Not Found
[2024-04-16T09:48:50.770+0000] {python.py:237} INFO - Done. Returned value was: None
[2024-04-16T09:48:50.771+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-04-16T09:48:50.780+0000] {taskinstance.py:1205} INFO - Marking task as SUCCESS. dag_id=spotify_final_dag, task_id=hello2, execution_date=20240416T063000, start_date=20240416T094849, end_date=20240416T094850
[2024-04-16T09:48:50.810+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2024-04-16T09:48:50.828+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-04-16T09:48:50.830+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
